acme/acme/client.py

```diff
@@ -624,7 +624,32 @@ def _send_request(self, method, url, *args, **kwargs):
         kwargs.setdefault('headers', {})
         kwargs['headers'].setdefault('User-Agent', self.user_agent)
         kwargs.setdefault('timeout', self._default_timeout)
-        response = self.session.request(method, url, *args, **kwargs)
+        try:
+            response = self.session.request(method, url, *args, **kwargs)
+        except requests.exceptions.RequestException as e:
+            """ Regex for parsing error messages"""
+            """ Regular expression groups"""
+            """ Match Group #1: Attempted connection host"""
+            """ Match Group #2: urllib3 Error Number"""
+            """ Match Group #3: urllib3 Error String"""
+            """ Example:"""
+            """HTTPSConnectionPool(host='acme-v01.api.letsencrypt.org', 
+            port=443): Max retries exceeded with url: /directory 
+            (Caused by NewConnectionError('
+            <requests.packages.urllib3.connection.VerifiedHTTPSConnection
+            object at 0x108356c50>: Failed to establish a new connection:
+            [Errno 65] No route to host',))"""
+            """ Match Group #1: acme-v01.api.letsencrypt.org"""
```

Comment:
Similarly, you don't need to provide an example of each matched group. The single example error message, plus the names assigned to the match groups, are enough.